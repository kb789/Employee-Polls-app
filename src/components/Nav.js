import { Link, useLocation } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { setUser } from "../features/currUserSlice";

const Nav = () => {
  const dispatch = useDispatch();

  const nonactive = "text-gray-500";

  const active = "text-black font-bold";

  const location = useLocation();

  return (
    <>
      <nav className="flex justify-center pt-2 pb-12 space-x-10 text-md uppercase">
        <Link to="/" className={location.pathname === "/" ? active : nonactive}>
          Home
        </Link>
        <span className="px-.5">|</span>
        <Link
          to="/add"
          className={location.pathname === "/add" ? active : nonactive}
        >
          Add Poll
        </Link>
        <span className="px-.5">|</span>
        <Link
          to="/leaderboard"
          className={location.pathname === "/leaderboard" ? active : nonactive}
        >
          Leaderboard
        </Link>
        <span className="px-.5">|</span>
        <button onClick={() => dispatch(setUser(""))}>
          <span className="text-md uppercase text-gray-500 ">Logout</span>
        </button>
      </nav>
    </>
  );
};
export default Nav;
