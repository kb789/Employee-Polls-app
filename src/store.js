import { configureStore } from "@reduxjs/toolkit";
import currUserReducer from "./features/currUserSlice";
import currAnsweredReducer from "./features/currAnsweredSlice";
import currPageReducer from "./features/currPageSlice";
import allQuestionsReducer from "./features/questionsSlice";
export const store = configureStore({
  reducer: {
    currUser: currUserReducer,
    currAnswered: currAnsweredReducer,
    currPage: currPageReducer,
    questions: allQuestionsReducer,
  },
});
