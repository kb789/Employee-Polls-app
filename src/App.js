import Nav from "./components/Nav";
import { Routes, Route } from "react-router-dom";

import { useDispatch, useSelector } from "react-redux";
import Home from "./components/Home";
import Add from "./components/Add";
import Leaderboard from "./components/Leaderboard";
import Login from "./components/Login";
import NotFound from "./components/NotFound";

import Question from "./components/Question";

import { getQuestions } from "./features/questionsSlice";
import { getUsers } from "./features/currUserSlice";
import { useEffect } from "react";

function App() {
  const { currUser, isLoadingUsers } = useSelector((store) => store.currUser);
  const { questions, isLoadingQuestions } = useSelector(
    (store) => store.questions
  );

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getUsers());
    dispatch(getQuestions());
  }, []);

  if (currUser === "") {
    return <Login />;
  }

  return (
    <main>
      <Nav />
      <Routes>
        <Route path="*" element={<NotFound />} />
        <Route path="/" on exact element={<Home />}></Route>
        <Route path="/add" exact element={<Add />}></Route>
        <Route path="/leaderboard" exact element={<Leaderboard />}></Route>
        <Route path="/login" exact element={<Login />}></Route>

        <Route path="/questions/:id" element={<Question />}></Route>
      </Routes>
    </main>
  );
}
export default App;
