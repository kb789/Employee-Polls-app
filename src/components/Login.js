import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { setUser, toggleisUserSignUp } from "../features/currUserSlice";
import Home from "./Home";
import Signup from "./Signup";

const Login = () => {
  const dispatch = useDispatch();

  const { currUser, users, isUserSignUp } = useSelector(
    (store) => store.currUser
  );

  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [loginError, setLoginError] = useState(false);

  const handleSignUp = (event) => {
    dispatch(toggleisUserSignUp());
  };

  const handleUsername = (event) => {
    setUsername(event.target.value);
  };

  const handlePassword = (event) => {
    setPassword(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    if (username in users && users[username]["password"] === password) {
      setLoginError(false);
   
      dispatch(setUser(users[username]));

      
    } else {
      setLoginError(true);
    }
  };

  if (currUser !== "") {
    return <Home />;
  }

  if (isUserSignUp) {
    return <Signup />;
  }
  return (
    <div>
      <h1 className="font-family: ui-sans-serif text-center text-3xl mb-8 mt-4 font-bold tracking-tight text-gray-900">
        Employee Polls
      </h1>
      <h1 className="text-center text-xl mb-4">Login</h1>

      <form
        className="mx-auto max-w-md space-y-6 bg-white py-8 px-10 rounded-lg shadow"
        onSubmit={handleSubmit}
      >
        {loginError ? (
          <h1 data-testid="er" className="text-center text-xl mb-4 text-red-500 font-semibold">
            Error: Incorrect Username or password.
          </h1>
        ) : (
          <div></div>
        )}
        Username:
        <input
          type="text"
          name="username"
          id="username"
          required
          onChange={handleUsername}
          data-testid="un"
          className="block w-full px-3 py-2 mt-6 border border-gray-300 
        rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
        focus:ring-blue-500 focus:border-blue-500 text-sm mb-4"
        />
        Password:
        <input
          type="password"
          name="password"
          id="password"
          required
          onChange={handlePassword}
          data-testid="pa"
          className="block w-full px-3 py-2 border border-gray-300 
        rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
        focus:ring-blue-500 focus:border-blue-500 text-sm"
        />
        <input
          type="submit"
          value="Submit"
          data-testid="submit"
          className="w-full flex justify-center mt-5 text-white bg-gradient-to-r from-cyan-400 via-cyan-500 to-cyan-600 hover:bg-gradient-to-br focus:ring-4 focus:outline-none focus:ring-cyan-300 dark:focus:ring-cyan-800 font-medium rounded-lg text-sm px-5 py-2.5 text-center mr-2 mb-2 "
        ></input>
      </form>
      <p className="text-center text-lg mb-4 mt-6 mr-2">
        Don't have an account?
        <button className="px-2"  data-testid="submit2" onClick={handleSignUp}>
          <span className="text-cyan-500"> Sign Up</span>
        </button>
      </p>
    </div>
  );
};
export default Login;
