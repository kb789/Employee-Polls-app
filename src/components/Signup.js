import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import {
  saveUser,
  setUser,
  toggleisUserSignUp,
  setUsers,
} from "../features/currUserSlice";
import Home from "./Home";

import Login from "./Login";
import Loading from "./Loading";

const Signup = () => {
  const dispatch = useDispatch();

  const { currUser, users, isLoadingUser, isUserSignUp } =
    useSelector((store) => store.currUser);

  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [fullname, setFullname] = useState("");
  const [avatar, setAvatar] = useState("");
  const [signUpError, setSignUpError] = useState(false);

  const handleLogin = (event) => {
    dispatch(toggleisUserSignUp());
  };

  const handleUsername = (event) => {
    setUsername(event.target.value);
  };

  const handlePassword = (event) => {
    setPassword(event.target.value);
  };

  const handleFullname = (event) => {
    setFullname(event.target.value);
  };

  const handleAvatar = (event) => {
    setAvatar(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    let userIndex = username in users;

    if (userIndex === true) {
      setSignUpError(true);
    } else {
      setSignUpError(false);
      dispatch(
        saveUser({
          id: username,
          password: password,
          name: fullname,
          avatarURL: avatar,
        })
      )
        .then((response) => {
          
          dispatch(setUsers(response.payload.users));
          dispatch(setUser(response.payload.newuser));
         
        })
        
         
    }
  };

  if (isLoadingUser) {
    return <Loading />;
  }

  if (!isUserSignUp) {
    return <Login />;
  }

  if (currUser !== "") {
    return <Home />;
  }

  return (
    <div>
      <h1 className="font-family: ui-sans-serif text-center text-3xl mb-8 mt-4 font-bold tracking-tight text-gray-900">
        Employee Polls
      </h1>
      <h1 className="text-center text-xl mb-4">Sign Up</h1>

      <form
        className="mx-auto max-w-md space-y-6 bg-white py-8 px-10 rounded-lg shadow"
        onSubmit={handleSubmit}
      >
        {signUpError ? (
          <h1 className="text-center text-xl mb-4 text-red-500 font-semibold">
            Error: That username is taken. Please sign up with a different one.
          </h1>
        ) : (
          <div></div>
        )}
        Username:
        <input
          type="text"
          name="username"
          id="username"
          required
          onChange={handleUsername}
          className="block w-full px-3 py-2 mt-6 border border-gray-300 
            rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
            focus:ring-blue-500 focus:border-blue-500 text-sm mb-4"
        />
        Password:
        <input
          type="password"
          name="password"
          id="password"
          required
          onChange={handlePassword}
          className="block w-full px-3 py-2 border border-gray-300 
            rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
            focus:ring-blue-500 focus:border-blue-500 text-sm"
        />
        Name (first and last)
        <input
          type="text"
          name="fullname"
          id="fullname"
          required
          onChange={handleFullname}
          className="block w-full px-3 py-2 mt-6 border border-gray-300 
            rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
            focus:ring-blue-500 focus:border-blue-500 text-sm mb-4"
        />
        Avatar URL
        <input
          type="text"
          name="avatar"
          id="avatar"
          required
          onChange={handleAvatar}
          className="block w-full px-3 py-2 mt-6 border border-gray-300 
            rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
            focus:ring-blue-500 focus:border-blue-500 text-sm mb-4"
        />
        <input
          type="submit"
          value="Submit"
          className="w-full flex justify-center mt-5 text-white bg-gradient-to-r from-cyan-400 via-cyan-500 to-cyan-600 hover:bg-gradient-to-br focus:ring-4 focus:outline-none focus:ring-cyan-300 dark:focus:ring-cyan-800 font-medium rounded-lg text-sm px-5 py-2.5 text-center mr-2 mb-2 "
        ></input>
      </form>
      <p className="text-center text-lg mb-4 mt-6 mr-2">
        Already have an account?
        <button className="px-2" onClick={handleLogin}>
          <span className="text-cyan-500">Login</span>
        </button>
      </p>
    </div>
  );
};

export default Signup;
