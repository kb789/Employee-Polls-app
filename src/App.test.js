import * as React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Nav from "./components/Nav";
import Login from "./components/Login";
import Signup from "./components/Signup";
import App from "./App";
import { Provider } from "react-redux";
import { store } from "./store";
import { MemoryRouter } from "react-router";

const data = require("./_DATA");

describe("_saveQuestion", () => {
  it("will return the saved question if question is saved", async () => {
    var result = await data._saveQuestion({
      optionOneText: "one",
      optionTwoText: "two",
      author: "id",
    });
    expect(result.optionOne.text).toMatch("one");
    expect(result.optionTwo.text).toMatch("two");
    expect(result.author).toMatch("id");
  });
  it("will return an error if the question is not saved", async () => {
    var badQuestion = {
      optionOneText: "one",
      author: "id",
    };
    await expect(data._saveQuestion(badQuestion)).rejects.toMatch(
      "Please provide optionOneText, optionTwoText, and author"
    );
  });
});

describe("_saveQuestionAnswer", () => {
  it("will return true if the answer is saved", async () => {
    var result = await data._saveQuestionAnswer({
      authedUser: "zoshikanlu",
      qid: "vthrdm985a262al8qx3do",
      answer: "optionOne",
    });

    expect(result).toBe(true);
  });
  it("will return an error if the question answer is not saved", async () => {
    await expect(
      data._saveQuestionAnswer({
        authedUser: "notauser",
        qid: "questiondoesnotexist",
      })
    ).rejects.toMatch("Please provide authedUser, qid, and answer");
  });
});

describe("App", () => {
  it("matches the snapshot when user not logged in", () => {
    var view = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(view).toMatchSnapshot();
  });
});

describe("<Nav />", () => {
  it("Renders <Nav/> component correctly", () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Nav />
        </MemoryRouter>
      </Provider>
    );

    expect(screen.getByText(/HOME/i)).toBeInTheDocument();
    expect(screen.getByText(/ADD POLL/i)).toBeInTheDocument();
    expect(screen.getByText(/LEADERBOARD/i)).toBeInTheDocument();
    expect(screen.getByText(/LOGOUT/i)).toBeInTheDocument();
  });
});

describe("Login", () => {
  it("will display an error if the password is not submitted", () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      </Provider>
    );
    var add1 = screen.getByTestId("un");
    fireEvent.change(add1, { target: { value: "tylermcginnis" } });
    var submitButton = screen.getByTestId("submit");
    fireEvent.click(submitButton);
    expect(
      screen.getByText(/Error: Incorrect Username or password./i)
    ).toBeInTheDocument();
  });

  it("will display the signup page if the sign up button is clicked", () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      </Provider>
    );
    var submitButton2 = screen.getByTestId("submit2");
    fireEvent.click(submitButton2);
    expect(screen.getByText(/Sign Up/i)).toBeInTheDocument();
  });

});

describe("_saveUser", () => {
  it("will return an error if the user is not saved", async () => {
    var badQuestion = {
      name: "fullname",
      avatarURL: "avatar",
    };
    await expect(data._saveUser(badQuestion)).rejects.toMatch(
      "Please provide user id, password, name, and avatar url"
    );
  });
});

describe("Signup", () => {
  it("will display loading page is new user fills out sign up form", () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Signup />
        </MemoryRouter>
      </Provider>
    );
    var add1 = screen.getByTestId("un");
    fireEvent.change(add1, { target: { value: "newuser" } });
    var add2 = screen.getByTestId("pa");
    fireEvent.change(add2, { target: { value: "new" } });
    var add3 = screen.getByTestId("fn");
    fireEvent.change(add3, { target: { value: "new" } });
    var add4 = screen.getByTestId("av");
    fireEvent.change(add4, { target: { value: "new" } });
    var submitButton = screen.getByTestId("su");
    fireEvent.click(submitButton);
    expect(
      screen.getByTestId("loading")
    ).toBeInTheDocument();
  });

  
});