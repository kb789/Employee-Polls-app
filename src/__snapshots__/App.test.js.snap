// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot when user is logged in 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1
          class="font-family: ui-sans-serif text-center text-3xl mb-8 mt-4 font-bold tracking-tight text-gray-900"
        >
          Employee Polls
        </h1>
        <h1
          class="text-center text-xl mb-4"
        >
          Login
        </h1>
        <form
          class="mx-auto max-w-md space-y-6 bg-white py-8 px-10 rounded-lg shadow"
        >
          <div />
          Username:
          <input
            class="block w-full px-3 py-2 mt-6 border border-gray-300 
 rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
 focus:ring-blue-500 focus:border-blue-500 text-sm mb-4"
            data-testid="un"
            id="username"
            name="username"
            required=""
            type="text"
          />
          Password:
          <input
            class="block w-full px-3 py-2 border border-gray-300 
 rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
 focus:ring-blue-500 focus:border-blue-500 text-sm"
            data-testid="pa"
            id="password"
            name="password"
            required=""
            type="password"
          />
          <input
            class="w-full flex justify-center mt-5 text-white bg-gradient-to-r from-cyan-400 via-cyan-500 to-cyan-600 hover:bg-gradient-to-br focus:ring-4 focus:outline-none focus:ring-cyan-300 dark:focus:ring-cyan-800 font-medium rounded-lg text-sm px-5 py-2.5 text-center mr-2 mb-2 "
            data-testid="submit"
            type="submit"
            value="Submit"
          />
        </form>
        <p
          class="text-center text-lg mb-4 mt-6 mr-2"
        >
          Don't have an account?
          <button
            class="px-2"
            data-testid="submit2"
          >
            <span
              class="text-cyan-500"
            >
               Sign Up
            </span>
          </button>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1
        class="font-family: ui-sans-serif text-center text-3xl mb-8 mt-4 font-bold tracking-tight text-gray-900"
      >
        Employee Polls
      </h1>
      <h1
        class="text-center text-xl mb-4"
      >
        Login
      </h1>
      <form
        class="mx-auto max-w-md space-y-6 bg-white py-8 px-10 rounded-lg shadow"
      >
        <div />
        Username:
        <input
          class="block w-full px-3 py-2 mt-6 border border-gray-300 
 rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
 focus:ring-blue-500 focus:border-blue-500 text-sm mb-4"
          data-testid="un"
          id="username"
          name="username"
          required=""
          type="text"
        />
        Password:
        <input
          class="block w-full px-3 py-2 border border-gray-300 
 rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
 focus:ring-blue-500 focus:border-blue-500 text-sm"
          data-testid="pa"
          id="password"
          name="password"
          required=""
          type="password"
        />
        <input
          class="w-full flex justify-center mt-5 text-white bg-gradient-to-r from-cyan-400 via-cyan-500 to-cyan-600 hover:bg-gradient-to-br focus:ring-4 focus:outline-none focus:ring-cyan-300 dark:focus:ring-cyan-800 font-medium rounded-lg text-sm px-5 py-2.5 text-center mr-2 mb-2 "
          data-testid="submit"
          type="submit"
          value="Submit"
        />
      </form>
      <p
        class="text-center text-lg mb-4 mt-6 mr-2"
      >
        Don't have an account?
        <button
          class="px-2"
          data-testid="submit2"
        >
          <span
            class="text-cyan-500"
          >
             Sign Up
          </span>
        </button>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App matches the snapshot when user not logged in 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1
          class="font-family: ui-sans-serif text-center text-3xl mb-8 mt-4 font-bold tracking-tight text-gray-900"
        >
          Employee Polls
        </h1>
        <h1
          class="text-center text-xl mb-4"
        >
          Login
        </h1>
        <form
          class="mx-auto max-w-md space-y-6 bg-white py-8 px-10 rounded-lg shadow"
        >
          <div />
          Username:
          <input
            class="block w-full px-3 py-2 mt-6 border border-gray-300 
 rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
 focus:ring-blue-500 focus:border-blue-500 text-sm mb-4"
            data-testid="un"
            id="username"
            name="username"
            required=""
            type="text"
          />
          Password:
          <input
            class="block w-full px-3 py-2 border border-gray-300 
 rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
 focus:ring-blue-500 focus:border-blue-500 text-sm"
            data-testid="pa"
            id="password"
            name="password"
            required=""
            type="password"
          />
          <input
            class="w-full flex justify-center mt-5 text-white bg-gradient-to-r from-cyan-400 via-cyan-500 to-cyan-600 hover:bg-gradient-to-br focus:ring-4 focus:outline-none focus:ring-cyan-300 dark:focus:ring-cyan-800 font-medium rounded-lg text-sm px-5 py-2.5 text-center mr-2 mb-2 "
            data-testid="submit"
            type="submit"
            value="Submit"
          />
        </form>
        <p
          class="text-center text-lg mb-4 mt-6 mr-2"
        >
          Don't have an account?
          <button
            class="px-2"
            data-testid="submit2"
          >
            <span
              class="text-cyan-500"
            >
               Sign Up
            </span>
          </button>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1
        class="font-family: ui-sans-serif text-center text-3xl mb-8 mt-4 font-bold tracking-tight text-gray-900"
      >
        Employee Polls
      </h1>
      <h1
        class="text-center text-xl mb-4"
      >
        Login
      </h1>
      <form
        class="mx-auto max-w-md space-y-6 bg-white py-8 px-10 rounded-lg shadow"
      >
        <div />
        Username:
        <input
          class="block w-full px-3 py-2 mt-6 border border-gray-300 
 rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
 focus:ring-blue-500 focus:border-blue-500 text-sm mb-4"
          data-testid="un"
          id="username"
          name="username"
          required=""
          type="text"
        />
        Password:
        <input
          class="block w-full px-3 py-2 border border-gray-300 
 rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
 focus:ring-blue-500 focus:border-blue-500 text-sm"
          data-testid="pa"
          id="password"
          name="password"
          required=""
          type="password"
        />
        <input
          class="w-full flex justify-center mt-5 text-white bg-gradient-to-r from-cyan-400 via-cyan-500 to-cyan-600 hover:bg-gradient-to-br focus:ring-4 focus:outline-none focus:ring-cyan-300 dark:focus:ring-cyan-800 font-medium rounded-lg text-sm px-5 py-2.5 text-center mr-2 mb-2 "
          data-testid="submit"
          type="submit"
          value="Submit"
        />
      </form>
      <p
        class="text-center text-lg mb-4 mt-6 mr-2"
      >
        Don't have an account?
        <button
          class="px-2"
          data-testid="submit2"
        >
          <span
            class="text-cyan-500"
          >
             Sign Up
          </span>
        </button>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App matches the snapshot when user not logged in 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h1
          class="font-family: ui-sans-serif text-center text-3xl mb-8 mt-4 font-bold tracking-tight text-gray-900"
        >
          Employee Polls
        </h1>
        <h1
          class="text-center text-xl mb-4"
        >
          Login
        </h1>
        <form
          class="mx-auto max-w-md space-y-6 bg-white py-8 px-10 rounded-lg shadow"
        >
          <div />
          Username:
          <input
            class="block w-full px-3 py-2 mt-6 border border-gray-300 
 rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
 focus:ring-blue-500 focus:border-blue-500 text-sm mb-4"
            data-testid="un"
            id="username"
            name="username"
            required=""
            type="text"
          />
          Password:
          <input
            class="block w-full px-3 py-2 border border-gray-300 
 rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
 focus:ring-blue-500 focus:border-blue-500 text-sm"
            data-testid="pa"
            id="password"
            name="password"
            required=""
            type="password"
          />
          <input
            class="w-full flex justify-center mt-5 text-white bg-gradient-to-r from-cyan-400 via-cyan-500 to-cyan-600 hover:bg-gradient-to-br focus:ring-4 focus:outline-none focus:ring-cyan-300 dark:focus:ring-cyan-800 font-medium rounded-lg text-sm px-5 py-2.5 text-center mr-2 mb-2 "
            data-testid="submit"
            type="submit"
            value="Submit"
          />
        </form>
        <p
          class="text-center text-lg mb-4 mt-6 mr-2"
        >
          Don't have an account?
          <button
            class="px-2"
            data-testid="submit2"
          >
            <span
              class="text-cyan-500"
            >
               Sign Up
            </span>
          </button>
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1
        class="font-family: ui-sans-serif text-center text-3xl mb-8 mt-4 font-bold tracking-tight text-gray-900"
      >
        Employee Polls
      </h1>
      <h1
        class="text-center text-xl mb-4"
      >
        Login
      </h1>
      <form
        class="mx-auto max-w-md space-y-6 bg-white py-8 px-10 rounded-lg shadow"
      >
        <div />
        Username:
        <input
          class="block w-full px-3 py-2 mt-6 border border-gray-300 
 rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
 focus:ring-blue-500 focus:border-blue-500 text-sm mb-4"
          data-testid="un"
          id="username"
          name="username"
          required=""
          type="text"
        />
        Password:
        <input
          class="block w-full px-3 py-2 border border-gray-300 
 rounded-md shadow-sm placeholder-gray-400 focus:outline-none 
 focus:ring-blue-500 focus:border-blue-500 text-sm"
          data-testid="pa"
          id="password"
          name="password"
          required=""
          type="password"
        />
        <input
          class="w-full flex justify-center mt-5 text-white bg-gradient-to-r from-cyan-400 via-cyan-500 to-cyan-600 hover:bg-gradient-to-br focus:ring-4 focus:outline-none focus:ring-cyan-300 dark:focus:ring-cyan-800 font-medium rounded-lg text-sm px-5 py-2.5 text-center mr-2 mb-2 "
          data-testid="submit"
          type="submit"
          value="Submit"
        />
      </form>
      <p
        class="text-center text-lg mb-4 mt-6 mr-2"
      >
        Don't have an account?
        <button
          class="px-2"
          data-testid="submit2"
        >
          <span
            class="text-cyan-500"
          >
             Sign Up
          </span>
        </button>
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
